common/djangoapps/student/tests/test_models.py

```diff
@@ -197,3 +197,32 @@ def test_upgrade_deadline_with_schedule_and_professional_mode(self):
         ScheduleFactory(enrollment=enrollment)
         self.assertIsNotNone(enrollment.schedule)
         self.assertIsNone(enrollment.upgrade_deadline)
+
+
+class PendingEmailChangeTests(SharedModuleStoreTestCase):
+    """
+    Tests the deletion of PendingEmailChange records.
+    """
+    @classmethod
+    def setUpClass(cls):
+        super(PendingEmailChangeTests, cls).setUpClass()
+        cls.course = CourseFactory()
+        cls.user = UserFactory()
+        cls.user2 = UserFactory()
+
+    def setUp(self):
+        self.email_change, _ = PendingEmailChange.objects.get_or_create(
+            user=self.user,
+            new_email='new@example.com',
+            activation_key='a' * 32
+        )
+
+    def test_delete_by_user_happy_path(self):
+        record_was_deleted = PendingEmailChange.delete_by_user_value(self.user, field='user')
+        self.assertTrue(record_was_deleted)
+        self.assertEqual(0, len(PendingEmailChange.objects.all()))
+
+    def test_delete_by_user_does_nothing_for_user_with_no_pending_change(self):
```

Comment:
test_cannot_delete_user_with_no_pending_change?

Reply:
Yeah, but we're not deleting the user, we're deleting the pending email change :)

Reply:
Right, right. My main gripe is with "does nothing", which is a little vague. Yet again, consider this a nit.